apps/laboratory/src/components/Wagmi/WagmiConnectButton.tsx

```diff
@@ -37,13 +33,28 @@ const message = {
 export function WagmiConnectButton() {
   const toast = useToast()
   const { isConnected } = useAccount()
+  const { chain } = useNetwork()
   const { signMessageAsync } = useSignMessage({ message: 'Hello Web3Modal!' })
+  const domain = {
+    name: 'Ether Mail',
+    version: '1',
+    chainId: chain?.id,
+    verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC'
+  } as const
+
   const { signTypedDataAsync } = useSignTypedData({
     domain,
     message,
     primaryType: 'Mail',
     types
   })
+  const [connector, setConnector] = useState<string | null>(null)
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
There should be a specific case to use `setTimeout` with 0 sec value. Curious -- why we did this?

Reply:
It needs to run at the next event cycle, so that is why I put 0. Don't know if there is a cleaner solution